python/lsst/skymap/ringsSkyMap.py

```diff
@@ -98,14 +130,58 @@ def generateTract(self, index):
             ra, dec = 0, 0.5*math.pi
         else:
             dec = self._ringSize*(ringNum + 1) - 0.5*math.pi
-            ra = math.fmod(self.config.raStart + 2*math.pi*tractNum/self._ringNums[ringNum], 2*math.pi)
+            ra = ((2*math.pi*tractNum/self._ringNums[ringNum])*afwGeom.radians +
+                  self._raStart).wrap().asRadians()
 
         center = afwGeom.SpherePoint(ra, dec, afwGeom.radians)
         wcs = self._wcsFactory.makeWcs(crPixPos=afwGeom.Point2D(0, 0), crValCoord=center)
         return ExplicitTractInfo(index, self.config.patchInnerDimensions, self.config.patchBorder, center,
                                  0.5*self._ringSize*afwGeom.radians, self.config.tractOverlap*afwGeom.degrees,
                                  wcs)
 
+    def _decToRingNum(self, dec):
+        """Calculate ring number from Declination
+
+        Parameters
+        ----------
+        dec : `lsst.afw.geom.Angle`
+            Declination.
+
+        Returns
+        -------
+        ringNum : `int`
+            Ring number: -1 for the south polar cap, and increasing to the
+            north, ending with ``numRings`` for the north polar cap.
+        """
+        firstRingStart = self._ringSize*0.5 - 0.5*math.pi
+        if dec < firstRingStart:
+            # Southern cap
+            return -1
+        elif dec > firstRingStart*-1:
+            # Northern cap
+            return self.config.numRings
+        return int((dec.asRadians() - firstRingStart)/self._ringSize)
+
+    def _raToTractNum(self, ra, ringNum):
+        """Calculate tract number from the Right Ascension
+
+        Parameters
+        ----------
+        ra : `lsst.afw.geom.Angle`
+            Right Ascension.
+        ringNum : `int`
+            Ring number (from ``_decToRingNum``).
+
+        Returns
+        -------
+        tractNum : `int`
+            Tract number within the ring (starts at 0 for the tract at raStart).
+        """
```

Comment:
Please consider testing for "out of range". I realize it's a private method, but still...it's best to catch errors early.